src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1266 @@
+
+            self->scroll_areas[0].v_top = i;
+            ui::scrollview::update_thumbs(self, new_industries::widx::scrollview);
+        }
+
+        // 0x00458AAF
+        static void updateBuildableIndustries(window* self)
+        {
+            auto industryCount = 0;
+            for (uint16_t i = 0; i < objectmgr::get_max_objects(object_type::industry); i++)
+            {
+                auto industryObj = objectmgr::get<industry_object>(i);
+                if (industryObj == nullptr)
+                    break;
+                if (!is_editor_mode())
+                {
+                    if (!(industryObj->flags & industry_object_flags::can_be_founded_by_user))
+                        continue;
+                    if (current_year() < industryObj->designedYear)
+                        continue;
+                    if (current_year() > industryObj->obsoleteYear)
+                        continue;
+                }
+                self->row_info[industryCount] = i;
+                industryCount++;
+            }
+
+            self->var_83C = industryCount;
+            auto rowHover = -1;
+
+            if (_lastSelectedIndustry != 0xFF)
+            {
+                for (auto i = 0; i < self->var_83C; i++)
+                {
+                    if (_lastSelectedIndustry == self->row_info[i])
+                    {
+                        rowHover = _lastSelectedIndustry;
+                        break;
+                    }
+                }
+            }
+
+            if (rowHover == -1 && self->var_83C != 0)
+            {
+                rowHover = self->row_info[0];
+            }
+
+            self->row_hover = rowHover;
+            updateActiveThumb(self);
+        }
+
+        // 0x00457FFE
+        static void tabReset(window* self)
+        {
+            self->min_width = new_industries::window_size.width;
+            self->min_height = new_industries::window_size.height;
+            self->max_width = new_industries::window_size.width;
+            self->max_height = new_industries::window_size.height;
+            input::toolSet(self, common::widx::tab_new_industry, 40);
+
+            input::set_flag(input::input_flags::flag6);
+            common::showGridlines();
+            byte_E0C3D9 = 0;
+            dword_E0C39C = 0x80000000;
+
+            self->var_83C = 0;
+            self->row_hover = -1;
+            self->var_846 = -1;
+
+            updateBuildableIndustries(self);
+
+            _prng->rand_next();
+            _dword_E0C394 = _prng->srand_0();
+            _dword_E0C398 = _prng->srand_1();
+        }
+
+        // 0x004589E8
+        static void on_resize(window* self)
+        {
+            self->invalidate();
+            gfx::ui_size_t minWindowSize = { self->min_width, self->min_height };
+            gfx::ui_size_t maxWindowSize = { self->max_width, self->max_height };
+            self->set_size(minWindowSize, maxWindowSize);
+            updateActiveThumb(self);
```

Comment:
One of the issues is that this is being called every resize this should only called if prior to setting height self->height < minHeight or the self->height > maxHeight. There is still another issue though. Trying to work it out.

Reply:
Okay the other bug isn't due to you. There is a bug when you leave the scroll area and it resizes but it happens in master.